src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -504,9 +504,10 @@ protected function whereNested(Builder $query, $where)
         // Here we will calculate what portion of the string we need to remove. If this
         // is a join clause query, we need to remove the "on" portion of the SQL and
         // if it is a normal query we need to take the leading "where" of queries.
-        $offset = $query instanceof JoinClause ? 3 : 6;
+        $sql = $this->compileWheres($where['query']);
+        $offset = $query instanceof JoinClause && !str_starts_with($sql, 'where ') ? 3 : 6;
```

Comment:
alternative solution:
```suggestion
        $offset = $offset = $where['query'] instanceof JoinClause ? 3 : 6; ? 3 : 6;
```

not sure if this would break anything else?

but the code inside again checks the instance but this class and not of the $query variable